utils.go

```diff
@@ -51,15 +76,15 @@ func JoinTraceFromHeader(
 			if err != nil {
 				return nil, err
 			}
-			contextIDMap[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
+			carrier.TracerState[strings.TrimPrefix(key, ContextIDHTTPHeaderPrefix)] = unescaped
```

Comment:
re `X-`: http://tools.ietf.org/html/rfc6648

As for the other point: this file should probably go away. Specific header prefixes are an OT implementation issue, not an OT standards issue. (OT makes no guarantees about two communicating processes running different OT impls)